fds/utils.py

```diff
@@ -32,25 +32,11 @@ def execute_command(command: Union[str, List[str]], shell: bool = False, capture
         output = subprocess.run(command, shell=shell, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
     elif capture_output_and_write_to_stdout:
         output = subprocess.Popen(command, shell=shell, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-        stdout = []
-        stderr = []
-        while True:
-            reads = [output.stdout.fileno(), output.stderr.fileno()]
-            ret = select.select(reads, [], [])
-
-            for fd in ret[0]:
-                if fd == output.stdout.fileno():
-                    read = output.stdout.readline()
-                    sys.stdout.write(convert_bytes_to_string(read))
-                    stdout.append(read)
-                if fd == output.stderr.fileno():
-                    read = output.stderr.readline()
-                    sys.stderr.write(convert_bytes_to_string(read))
-                    stderr.append(read)
-            if output.poll() is not None:
-                break
+        stdout, stderr = output.communicate()
+        sys.stdout.write(convert_bytes_to_string(stdout))
+        sys.stderr.write(convert_bytes_to_string(stderr))
```

Comment:
The problem here is that the streams won't be outputted in correct order.
If there's an error written to `stderr` in the middle of output, it will still appear in the end in this way, and cause confusion.
Why not pass a splitting stream to `stdout` and `stderr`, which both stores the outputs in byte buffers, and writes them to `sys.stdout` and `sys.stderr` ?
Example: https://stackoverflow.com/a/616686

Reply:
Oh yeah, I forgot to do this part, We originally were doing it, I changed it so that it works in windows too